mismatched types error for x86_64-linux-android target

On travis, when building my project for `x86_64-linux-android` target, I got this error message:

```shell
Compiling net2 v0.2.32
error[E0308]: mismatched types
  --> /cargo/registry/src/github.com-1ecc6299db9ec823/net2-0.2.32/src/socket.rs:40:51
   |
40 |             ::cvt(c::bind(self.inner.raw(), addr, len as len_t)).map(|_| ())
   |                                                   ^^^^^^^^^^^^ expected u32, found i32
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -15,7 +15,7 @@ Extensions to the standard library's networking types as proposed in RFC 1158.
 winapi = { version = "0.3", features = ["handleapi", "winsock2", "ws2def", "ws2ipdef", "ws2tcpip"] }
 
 [target.'cfg(any(target_os="redox", unix))'.dependencies]
-libc = "0.2.41"
+libc = "0.2.42"
 
 [dependencies]
 cfg-if = "0.1"

--- a/src/socket.rs
+++ b/src/socket.rs
@@ -32,14 +32,9 @@ impl Socket {
     }
 
     pub fn bind(&self, addr: &SocketAddr) -> io::Result<()> {
-        #[cfg(not(all(any(target_arch = "aarch64", target_arch = "x86_64"), target_os = "android")))]
-        use sys::c::socklen_t as len_t;
-        #[cfg(all(any(target_arch = "aarch64", target_arch = "x86_64"), target_os = "android"))]
-        use libc::c_int as len_t;
-
         let (addr, len) = addr2raw(addr);
         unsafe {
-            ::cvt(c::bind(self.inner.raw(), addr, len as len_t)).map(|_| ())
+            ::cvt(c::bind(self.inner.raw(), addr, len as c::socklen_t)).map(|_| ())
         }
     }
 

